sklearn/tests/test_config.py

```diff
@@ -0,0 +1,26 @@
+from sklearn import get_config, set_config
+from sklearn.utils.testing import assert_equal
+
+
+def test_set_config():
+    assert_equal(get_config(), {'assume_finite': False})
+
+    # Not using as a context manager affects nothing
+    set_config(assume_finite=True)
```

Comment:
Explicit is better than implicit, so I'd say ``config_context``. Also it mirrors existing conventions more.
[double checking: @tacaswell are you happy with your naming? of config context managers?]

If you really prefer ``with config(...)`` I wouldn't protest too much ;)
I would add ``set_config``, too.